mm2src/mm2_main/src/lp_native_dex.rs

```diff
@@ -549,21 +487,76 @@ fn get_p2p_key(ctx: &MmArc, i_am_seed: bool) -> P2PResult<[u8; 32]> {
     Ok(p2p_key)
 }
 
-pub async fn init_p2p(ctx: MmArc) -> P2PResult<()> {
-    let i_am_seed = ctx.is_seed_node();
+fn p2p_precheck(ctx: &MmArc) -> P2PResult<()> {
```

Comment:
small improvement for errors handling in p2p precheck

we could add private func or closure inside `p2p_precheck` which builds and returns necessary err and makes code a bit cleaner

```rust
    let precheck_err = |reason: &str| {
        MmError::err(P2PInitError::Precheck {
            reason: reason.to_owned(),
        })
    };

    let seednodes = seednodes(ctx)?;

    if is_bootstrap_node && !is_seed_node {
        return precheck_err("Bootstrap node must also be a seed node.");
    }
```

Reply:
Makes sense, thanks.